### DISCUSSION TITLE
Grouping updates by sub-directories

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#28125] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have a monorepo with a directory structure for my IaC as follows:

```
terraform/environments
├── aws-account-1
│   └── region-1
│       └── environment-1
└── aws-account-2
    └── region-1
        ├── environment-2
        └── environment-3
    └── region-2
        └── environment-4
```

I'd like to group my provider updates by both provider name and also by account. With the above dummy directory structure I'd like the following PRs when my providers need updating, e.g. with the `aws` provider:

* aws-account-1: Update Terraform aws to 1.2.3
* aws-account-2: Update Terraform aws to 1.2.3

How can I do this? Would I need to make one group per AWS account and then interpolate the `groupSlug` somewhere?

### Logs (if relevant)

N/A; or please advise what would be useful.

**Comment by rarkins** [comment]
Is `terraform/environments` meant to be the repo name or a nested directory?

**Reply by allanlewis** [reply]
It's a directory name, the repo name isn't shown in the tree.

**Reply by rarkins** [reply]
So you want to group by a directory which is part way down through the tree?

**Reply by allanlewis** [reply]
Yes, that's basically it. I was thinking maybe there's a way to group the updates from a given subdirectory (recursive) but then split by dependency name.

**Reply by rarkins** [reply]
You have a structure like a/b/c/d/e and want to group by c, so that's pretty hard.

First thing to remember is that "grouping" simply means that updates have the same branch name. In this question you aren't so much asking how to group updates but how to _split_ grouped updates.

Splitting is best done by using `additionaBranchPrefix` with templating. This previous discussion is similar to yours: https://github.com/renovatebot/renovate/discussions/26968

Except if you want it done by a mid-level directory then you'll need to experiment with the handlebars "replace" directive to massage the string to get just the part you want

**Reply by allanlewis** [reply]
Thanks, @rarkins - I appreciate your help with this. I'll look into using the approach you suggested.

**Reply by rarkins** [reply, marked as answer]
Another approach would be to use matchFileNames patterns + groupName and have a different groupName for each account. If you have a small number of accounts then it can be manageable to add these rules manually instead of trying to do a generic solution as I described earlier with additionalBranchPrefix and templates

**Reply by allanlewis** [reply]
> If you have a small number of accounts...

Indeed I do, so this might be a simpler approach - thanks again!
